### PR TITLE
added phone_production exercise and it's test

### PR DESCRIPTION
**Pull request by AGuyNextDoor**

### CODE REVIEW

File: 02_programming_fundamentals/08_this_day_1/05_phone_production/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "jest": {
+    "noStackTrace": true
+  },
+  "dependencies": {
+    "jest": "^23.6.0",
+    "shelljs": "^0.8.2"
+  }
```
Comment (Fenntasy) [inline 02_programming_fundamentals/08_this_day_1/05_phone_production/package.json]: When the `test` script starts with `eslint .`, you need to add 
```
  "eslintIgnore": [
    "__tests__"
  ]
```

to the `package.json` for `eslint` to ignore the test directory

**Review by Fenntasy — COMMENTED** [review]
I'm not sure I undertstand, in the exercise description you say

> In this exercise you are given an `phone` object that possesses the basic structure of a cellphone.

So I assumed I was going to find something in the exercise file. And if I try with the file you put in `sparta_courses`, the tests are failing.

Is it not supposed to be the basis for this file?

**Comment by AGuyNextDoor** [comment]
https://github.com/fewlinesco/sparta_courses/pull/317

**Comment by AGuyNextDoor** [comment]
most probably an information I said I was going to give when writting the description and that didn't follow through.. 

Some / Most of the work till now is a first "draft", that probably hasn't been reread and were meant to, they all probably possesses small contradictions because they have not been yet completed as they were planned to.. Be gentle ^^

File: 02_programming_fundamentals/08_this_day_1/05_phone_production/__tests__/phone_production.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+beforeAll(() => {
+  // Loads the student's code
+  studentCode = readcode(path.resolve(__dirname, "../05_phone_production.js"));
+  return studentCode;
+});
+
+test("Testing if massung has the right properties", () => {
+  massung.functional = true;
+  massung.autodestruction();
+  expect(massung).toHaveProperty('autodestruction');
```
Comment (Fenntasy) [inline 02_programming_fundamentals/08_this_day_1/05_phone_production/__tests__/phone_production.test.js]: It's a detail but for those who are trying to go step by step, the order of the verifications can feel weird.
You're testing first `autodestruction` but that's the last thing you asked for in the exercise.

It's something that I would not advise in a real program but for these tests, could you make a test per property? That way, students see progress :)

File: 02_programming_fundamentals/08_this_day_1/05_phone_production/__tests__/phone_production.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+test("Testing if jPhone has the right properties", () => {
+  jPhone.turnedOn = true;
+  jPhone.batteryObsolescence();
+  expect(jPhone).toHaveProperty('batteryObsolescence');
+  expect(jPhone.system).toMatch("JOS");
+  expect(jPhone.screenSize).toBe(6.5);
+  expect(jPhone.turnedOn).toBe(false);
+});
+
+test("Testing if phone has the right insertSim function", () => {
```
Comment (Fenntasy) [inline 02_programming_fundamentals/08_this_day_1/05_phone_production/__tests__/phone_production.test.js]: This test should be the first one in the file for the same reason as previously, to see progress when following the exercise.

**Review by Fenntasy — APPROVED** [review]

**Pull request merged**